react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { index: 1 };
+
+    // bind here
+    this.changeIndex = this.changeIndex.bind(this);
+    this.changeIndexArrow = this.changeIndexArrow.bind(this);
+    this.changeIndexCircle = this.changeIndexCircle.bind(this);
+    this.returnImage = this.returnImage.bind(this);
+    this.returnCirclePos = this.returnCirclePos.bind(this);
+  }
+
+  // do not need to bind lifecyle methods
+  componentDidMount() {
+    this.timerId = setInterval(
+      () => this.changeIndex(),
+      1000
+    );
+  }
+
+  // do not need to bind lifecyle methods
+  componentWillUnmount() {
+    clearInterval(this.timerId);
+  }
+
+  changeIndex() {
+    if (this.state.index === 5) {
+      this.setState({ index: 1 });
+    } else {
+      this.setState({ index: this.state.index + 1 });
+    }
+  }
+
+  changeIndexArrow(event) {
+    if (event.target.className === 'fas fa-chevron-circle-left arrow') {
+      if (this.state.index === 1) {
+        this.setState({ index: 5 });
```

Comment:
Using `5` here is a "magic" number, what if someone passes in an array with 14 images in it? It would be better to use `this.props.images.length - 1` to go to the end of the array